platform/kubernetes/kubernetes.go

```diff
@@ -435,6 +437,79 @@ func (c *Cluster) Version() (string, error) {
 	return c.version, nil
 }
 
+func (c *Cluster) Export() ([]byte, error) {
+	var config bytes.Buffer
+	list, err := c.client.Namespaces().List(api.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "getting namespaces")
+	}
+	for _, ns := range list.Items {
+		err := appendYAML(&config, "v1", "Namespace", ns)
+		if err != nil {
+			return nil, errors.Wrap(err, "marshalling namespace to YAML")
+		}
+
+		deployments, err := c.client.Deployments(ns.Name).List(api.ListOptions{})
+		if err != nil {
+			return nil, errors.Wrap(err, "getting deployments")
+		}
+		for _, deployment := range deployments.Items {
+			if isAddon(&deployment) {
+				continue
+			}
+			err := appendYAML(&config, "extensions/v1beta1", "Deployment", deployment)
+			if err != nil {
+				return nil, errors.Wrap(err, "marshalling deployment to YAML")
+			}
+		}
+
+		rcs, err := c.client.ReplicationControllers(ns.Name).List(api.ListOptions{})
+		if err != nil {
+			return nil, errors.Wrap(err, "getting replication controllers")
+		}
+		for _, rc := range rcs.Items {
+			if isAddon(&rc) {
+				continue
+			}
+			err := appendYAML(&config, "v1", "ReplicationController", rc)
+			if err != nil {
+				return nil, errors.Wrap(err, "marshalling replication controller to YAML")
+			}
+		}
+
+		services, err := c.client.Services(ns.Name).List(api.ListOptions{})
+		if err != nil {
+			return nil, errors.Wrap(err, "getting services")
+		}
+		for _, service := range services.Items {
+			if isAddon(&service) {
+				continue
+			}
+			err := appendYAML(&config, "v1", "Service", service)
+			if err != nil {
+				return nil, errors.Wrap(err, "marshalling service to YAML")
+			}
+		}
+	}
```

Comment:
Ah OK, just sticking with deployments, namespaces and services for now, that explains the cases when picking a short name, above. We can add to these later, although perhaps we should talk about protocol changes first.